lock-impl/src/main/java/com/palantir/lock/impl/LegacyTimelockService.java

```diff
@@ -186,18 +173,72 @@ public long currentTimeMillis() {
 
     @Override
     public TimestampLeaseResults acquireTimestampLeases(Map<TimestampLeaseName, Integer> requests) {
-        // TODO(aalouane): implement!
-        throw new UnsupportedOperationException("Not implemented yet!");
+        // This implementation here treats ALL the timestamp leases the same, meaning they will all progress
+        // at the same rate. This is only a perf concern (all timestamps progress together), not correctness.
+        // TODO(jakubk): This should reject timestamp lease names that conflict with the immutable timestamp lease
+        // or we should fix the implementation in timelock to allow these to conflict.
+        return withLeaseFreshTimestamp(TIMESTAMP_LEASES_CLIENT_ID.get(), response -> {
+            Map<TimestampLeaseName, TimestampLeaseResult> results = EntryStream.of(requests)
+                    .mapValues(numFreshTimestamps -> {
+                        TimestampRange timestampRange = getFreshTimestamps(numFreshTimestamps);
+                        Preconditions.checkState(
+                                timestampRange.size() >= numFreshTimestamps, "Didn't get enough timestamps");
```

Comment:
No need for metrics here?

Reply:
At least for now, I am comfortable with this implementation being bad and holding it as "just for tests". I'll FLUP with what do we do about LegacyTimelockService and where is it used in general, and if it turns out that we should raise the bar on the implementation quality here, I can FLUP with that.